agent/src/main/java/org/openremote/agent/protocol/http/HTTPProtocol.java

```diff
@@ -602,4 +613,68 @@ protected void cancelPolling(AttributeRef attributeRef) {
         }
     }
 
+	public static class DynamicTimeReplacementClientRequestFilter implements ClientRequestFilter {
+
+		private final TimerService timerService;
+
+		private DynamicTimeReplacementClientRequestFilter(TimerService timerService) {
+			this.timerService = timerService;
+		}
+		@Override
+		public void filter(ClientRequestContext requestContext) throws IOException {
+			if (requestContext.hasEntity()) {
+				Object entity = requestContext.getEntity();
+
+				if (entity instanceof InputStream) {
+					String body = new Scanner((InputStream) entity, StandardCharsets.UTF_8).useDelimiter("\\A").next();
+
+					// Perform regex replacement
+					String modifiedBody = replace(body);
+
+					// Set the modified body back to the request context
+					InputStream modifiedStream = new ByteArrayInputStream(modifiedBody.getBytes(StandardCharsets.UTF_8));
+					// Set the modified body back to the request context
+					requestContext.setEntity(Entity.entity(modifiedBody, MediaType.APPLICATION_JSON));
+				} else if (entity instanceof String) {
+					// Perform regex replacement
+					String modifiedBody = replace((String) entity);
+
+					// Set the modified body back to the request context
+					requestContext.setEntity(modifiedBody);
+				}
+			}
+		}
+
+		private String replace(String str){
+			// Regex to match the pattern ${time:DURATION:FORMAT}
+			String regex = "\\$\\{time:([^:]+):([^}]+)\\}";
```

Comment:
`DURATION` and `FORMAT` should be optional so a regex like: `\\$\\{time(?::([^:]+))?(?::([^}]+))?\\}`

Default for `DURATION` should just be `now()`
Default for `FORMAT` should just be epoch seconds